app/queries/invoices_query.rb

```diff
@@ -18,20 +19,22 @@ def call(search_term:, status:, payment_status:, page:, limit:, filters: {}) # r
 
   attr_reader :search_term
 
-  def base_scope
-    organization.invoices.ransack(search_params)
+  def base_scope(customer_id:)
+    organization.invoices.ransack(search_params(customer_id))
   end
 
-  def search_params
+  def search_params(customer_id)
     return nil if search_term.blank?
 
-    {
-      m: 'or',
-      id_cont: search_term,
-      number_cont: search_term,
-      customer_name_cont: search_term,
-      customer_external_id_cont: search_term,
-      customer_email_cont: search_term,
-    }
+    {}.tap do |h|
+      h[:m] = 'or'
+      h[:id_cont] = search_term
+      h[:number_cont] = search_term
+      unless customer_id
+        h[:customer_name_cont] = search_term
+        h[:customer_external_id_cont] = search_term
+        h[:customer_email_cont] = search_term
+      end
```

Comment:
This approach feels a bit more rubist ;) :

```suggestion
    terms = {
      m: 'or',
      id_cont: search_term,
      number_cont: search_term,
    }
    return terms if customer_id.present?
    
    terms.merge(
     customer_name_cont: search_term,
     customer_external_id_cont: search_term,
     customer_email_cont: search_term,
    )
```

Reply:
code updated